etk-asm/src/asm.rs

```diff
@@ -184,10 +218,14 @@ impl Assembler {
     pub fn finish(self) -> Result<(), Error> {
         if let Some(undef) = self.pending.front() {
             return match undef {
-                RawOp::Op(op) => {
+                RawOp::Op(AbstractOp::Op(op)) => {
                     let label = op.immediate_label().unwrap();
                     error::UndeclaredLabel { label }.fail()
                 }
+                RawOp::Op(AbstractOp::Macro(m)) => error::UndeclaredMacro {
+                    macro_name: &m.name,
+                }
+                .fail(),
                 _ => unreachable!(),
```

Comment:
You might already know, but I just wanted to mention that this will get hit for variable-sized pushes if the label is undefined. You might need to add patterns for the other `AbstractOp` variants?

Reply:
Ah good catch, thanks.